src/create-theme-provider.js

```diff
@@ -24,25 +23,53 @@ export default function createThemeProvider<Theme>(context: Context<Theme>) {
     getTheme(outerTheme: Theme) {
       const { theme } = this.props;
 
-      if (typeof theme === 'function') {
-        const mergedTheme = theme(outerTheme);
+      // Check if any of the themes have changed or if it we don't have a cached theme yet
```

Comment:
This comment describes what it does, even though its obvious. Instead comments should explain why. If thats obvious comments are not needed.